src/charm.py

```diff
@@ -0,0 +1,208 @@
+#!/usr/bin/env python3
+import glob
+import subprocess
+import time
+
+from jinja2 import Environment, FileSystemLoader
+import logging
+from pathlib import Path
+
+from kubernetes import client, config
+import kubernetes.client.exceptions
+from oci_image import OCIImageResource
+from ops.charm import CharmBase
+from ops.framework import StoredState
+from ops.main import main
+from ops.model import ActiveStatus, WaitingStatus, MaintenanceStatus
+
+
+class MetacontrollerOperatorCharm(CharmBase):
+    """Charm the Metacontroller"""
+
+    _stored = StoredState()
+
+    def __init__(self, *args):
+        super().__init__(*args)
+
+        if not self.unit.is_leader():
+            self.model.unit.status = WaitingStatus("Waiting for leadership")
+            return
+
+        self.framework.observe(self.on.noop_pebble_ready, self._noop_pebble_ready)
+        self.framework.observe(self.on.install, self._install)
+        self.framework.observe(self.on.remove, self._remove)
+        self.framework.observe(self.on.update_status, self._update_status)
+        # self.framework.observe(self.on.config_changed, self._reconcile)
+
+        self.x = 1
+        self.logger = logging.getLogger(__name__)
+        self.y = 1
+        # TODO: Fix file imports and move ./src/files back to ./files
+        self._manifests_file_root = None
+        self.manifest_file_root = "./src/files/manifests/"
+        self.image = OCIImageResource(self, "oci-image")
+
+    def _noop_pebble_ready(self, _):
+        self.logger.info("noop_pebble_ready fired")
+
+    def _install(self, event):
+        self.logger.info("Installing by instantiating Kubernetes objects")
+        self.unit.status = MaintenanceStatus("Instantiating Kubernetes objects")
+        _, manifests_str = self._render_manifests()
+
+        self.logger.info("Applying manifests")
+
+        completed_process = subprocess.run(
+            ["./kubectl", "apply", "-f-"],
+            input=manifests_str.encode("utf-8"),
+            stderr=subprocess.STDOUT,
+            stdout=subprocess.PIPE,
+        )
```

Comment:
We would probably want to move this to lightkube instead of kubectl if it isn't too much work.

Reply:
Agreed.  Will do so on the next pass